app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/MapsActivity.java

```diff
@@ -92,6 +112,116 @@ public void onSearchTextChanged(String oldQuery, String newQuery) {
+                        LocationAutocomplete suggestion = (LocationAutocomplete) item;
+
+                        if (suggestion.getPlace().getPlaceID() == null) {
+                            leftIcon.setImageDrawable(ResourcesCompat.getDrawable(getResources(),
+                                    R.drawable.ic_bus_stop, null));
+                            textView.setTextColor(Color.parseColor("#111111"));
+                            textView.setTextSize(15f);
+                        } else {
+                            leftIcon.setImageDrawable(ResourcesCompat.getDrawable(getResources(),
+                                    R.drawable.ic_loc_stop, null));
+                            Util.setIconColor(leftIcon, Color.parseColor("#cacaca"));
+                            String text = "<font color='black'>" + suggestion.getPlace().getName()
+                                    + "</font>"
+                                    + "<br/>" + suggestion.getPlace().getDetail();
+                            textView.setText(Html.fromHtml(text), TextView.BufferType.SPANNABLE);
+                            textView.setTextSize(15f);
+                        }
+                    }
+
+                });
+
+        mSearchView.setOnSearchListener(new FloatingSearchView.OnSearchListener() {
+            @Override
+            public void onSuggestionClicked(SearchSuggestion searchSuggestion) {
+                final Place dest = ((LocationAutocomplete) searchSuggestion).getPlace();
+                new Thread(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (dest.getPlaceID() == null) {
+                            launchRoute(MapsActivity.this.lastLocation.getLatitude() +
+                                            ", " + MapsActivity.this.lastLocation.getLongitude(),
+                                    dest.toString(), dest.getName());
+                        } else {
+                            FetchPlaceRequest request = FetchPlaceRequest.builder(dest.toString(),
+                                    Arrays.asList(
+                                            com.google.android.libraries.places.api.model.Place.Field.LAT_LNG)).build();
+
+                            placesClient.fetchPlace(request).addOnSuccessListener((response) -> {
+                                launchRoute(MapsActivity.this.lastLocation.getLatitude() +
+                                                ", " + MapsActivity.this.lastLocation.getLongitude(),
+                                        response.getPlace().getLatLng().latitude + ", "
+                                                + response.getPlace().getLatLng().longitude,
+                                        dest.getName());
+
+                            });
+                        }
+                    }
+                }).start();
+            }
+
+            @Override
+            public void onSearchAction(String currentQuery) {
```

Comment:
does this override do anything?

Reply:
Nope, but I can't remove it without OnSearchListener acting up, so it's staying there for the time being.